x-pack/filebeat/input/azureeventhub/sanitizers.go

```diff
@@ -0,0 +1,216 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+
+package azureeventhub
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"regexp"
+)
+
+const (
+	SanitizerNewLines     = "new_lines"
+	SanitizerSingleQuotes = "single_quotes"
+	SanitizerReplaceAll   = "replace_all"
+)
+
+// ----------------------------------------------------------------------------
+// Sanitizer API
+// ----------------------------------------------------------------------------
+
+// SanitizerSpec defines a sanitizer configuration.
+//
+// Sanitizers can use the spec field to provide additional
+// configuration.
+type SanitizerSpec struct {
+	Type string                 `config:"type"`
+	Spec map[string]interface{} `config:"spec"`
+}
```

Comment:
+1 on this. This gives us flexibility to add custom sanitizer implementations :)